main.go

```diff
@@ -157,7 +157,11 @@ func (c *Controller) handleNewCSR(key string) error {
 	}
 	maxPending := getMaxPending(machines.Items)
 	atomic.StoreUint32(&maxPendingCSRs, uint32(maxPending))
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPending {
+	csrs := []*certificatesv1beta1.CertificateSigningRequest{}
+	for _, item := range c.indexer.List() {
+		csrs = append(csrs, item.(*certificatesv1beta1.CertificateSigningRequest))
+	}
+	if pending := recentlyPendingCSRs(csrs); pending > maxPending {
```

Comment:
What's the motivation for moving this outside of the function? I'm not a huge fan of moving logic around to suit the test suites, there are better ways we can do this, eg

```
type indexLister interface {
  List() []interface{}
}

func recentlyPendingCSRs(csrs indexLister) []*CertificateSigningRequest {
...
}

// In test code

type fakeIndexLister struct {
 items []interface{}
}

func (f fakeIndexList) List() ([]interface{}) {
  return f.items
}
```

Reply:
The goal of this PR is to transition to https://github.com/openshift/cluster-machine-approver/pull/73 and do not change existing tests. The thing you pointing out makes sense to me, but I didn't think about it as a long-term solution, just wanted to preserve tests.

Reply:
I don't think it would take too long to do, and would be the right thing given how long it might take to merge things across to that other PR, but it's not a blocker if that's going away in #73